src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -242,12 +242,8 @@ function WebGLShadowMap( _renderer, _objects, _capabilities ) {
 
 		}
 
-		if ( _renderer.localClippingEnabled &&
-				material.clipShadows === true &&
-				material.clippingPlanes.length !== 0 ||
-				material.displacementMap &&
-				material.displacementScale !== 0
-		) {
+		if ( ( _renderer.localClippingEnabled && material.clipShadows === true && material.clippingPlanes.length !== 0 ) ||
+			( material.displacementMap && material.displacementScale !== 0 ) || material.alphaMap ) {
```

Comment:
It worth checking if `material.alphaTest === 1` when there's an `alphaMap`? In that case we don't need to create or use a custom shadow material because no clipping will occur.

Reply:
I guess the check would then be...

```js
material.alphaTest !==0 && material.aphaTest !== 1 && material.alphaMap
```

Reply:
Done!

Reply:
Yup! And technically we don't need to render a mesh at all if alphaTest === 1 because every pixel would be discarded and ultimately nothing would be rendered but that might be an excessive optimization because the user should be hiding the mesh in that case, anyway.